Automattic/woocommerce-payments
# Dynamic gateway title by available payment methods

Fixes #2433 

#### Changes proposed in this Pull Request


# Dynamic Gateway Title and Some Extra Goodies

The focus of this PR is to update the title of the UPE payment gateway at checkout to make the title dependent on the number of payment methods enabled. If there is only one payment method enabled, the title of the gateway will just be the name of that payment method.

![Gateway title single method](https://cdn-std.droplr.net/files/acc_1104206/WN96Oc)
_Gateway title with only a single method enabled_

If there are two or more payment methods currently enabled, the title of the gateway will be named "Popular payment methods"

![Gateway title multiple methods](https://cdn-std.droplr.net/files/acc_1104206/EnxyG2)
_Gateway title with multiple methods enabled_

Also, note that there are now new tab selected styles: with multiple payment methods, when a payment method is selected, the tab will have a border/outline similar to the selected input states, instead of being shaded dark matching the pay button. Moreover fonts and styles for all text elements within the UPE should now be consistent with the existing theme. These changes should be reflected on all pages where the UPE is present, so this includes the following.
 - Class checkout page
 - Blocks checkout page
 - Pay for order page
 - Add payment method page
 - Change payment method page
 
Note that the UPE on the Blocks checkout in this PR is able to read theme styling from the stored transient, if this information has been saved earlier, but it cannot currently compute these styles on its own. This means that if you visit the regular checkout page before using the Blocks checkout, the Blocks checkout should be able to copy the stored themes; however, if you visit the Blocks checkout without any transient information available or with transient information for a different theme, the UPE will not reflect the current theme. This will be further explored and hopefully resolved in #2603.

In addition to all of the above, there are few changes here for the main WooCommerce payments settings screen (**WooCommerce > Settings > Payments > All payment methods**). The title of the payment gateway is now simply "WooCommerce Payments", with no subheading; the payment icons have been removed from this page; and the description has been updated to the following:
```
Payments made simple, with no monthly fees - designed exclusively for WooCommerce stores. Accept credit cards, debit cards, and other popular payment methods.
```

![New payments settings screen](https://cdn-std.droplr.net/files/acc_1104206/TyAOga)
_New general payments settings screen_




#### Testing instructions





1. Please pull the changes in this PR, the latest version of the Payments Server, and enable the UPE using the WCPay Dev Tools.
2. Visit the regular checkout with a single payment method and confirm that the gateway title is the name of that payment method.
3. Visit the regular checkout with more than one payment enabled and confirm that the gateway title is now "Popular payment methods".
4. Visit the regular checkout with multiple payment methods enabled and confirmed that the payment method tabs within the UPE are outlined when selected, instead of shaded. Also, confirm that all text within the UPE for all payment methods is formatted consistently.
5. Repeat these checks on all of the following pages:
 - Class checkout page
 - Blocks checkout page
 - Pay for order page
 - Add payment method page
 - Change payment method page
 6. Check the main WC Payments settings page (**WooCommerce > Settings > Payments > All payment methods**), confirm that the gateway title and description are correct, as described above. Confirm absence of payment method icons.
 7. One last step for good luck, so touch wood or something soft and warm, if wood unavailable.

-------------------
